core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts

```diff
@@ -3,52 +3,259 @@ import { MonacoEditorModule } from '@materia-ui/ngx-monaco-editor';
 import { byTestId, createComponentFactory, Spectator } from '@ngneat/spectator';
 
 import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { By } from '@angular/platform-browser';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 
 import { ButtonModule } from 'primeng/button';
 import { DialogModule } from 'primeng/dialog';
 
-import { DotMessageService } from '@dotcms/data-access';
+import { DotMessageService, DotUploadService } from '@dotcms/data-access';
+import { DotCMSTempFile } from '@dotcms/dotcms-models';
+import { DropZoneFileEvent } from '@dotcms/ui';
 
-import { BinaryFieldComponent } from './binary-field.component';
+import { DotBinaryFieldComponent } from './binary-field.component';
+import { DotUiMessageComponent } from './components/dot-ui-message/dot-ui-message.component';
+import {
+    BINARY_FIELD_MODE,
+    BINARY_FIELD_STATUS,
+    DotBinaryFieldStore
+} from './store/binary-field.store';
 
 import { CONTENTTYPE_FIELDS_MESSAGE_MOCK } from '../../utils/mock';
 
-describe('BinaryFieldComponent', () => {
-    let spectator: Spectator<BinaryFieldComponent>;
+const TEMP_FILE_MOCK: DotCMSTempFile = {
+    fileName: 'image.png',
+    folder: '/images',
+    id: '12345',
+    image: true,
+    length: 1000,
+    referenceUrl: '/reference/url',
+    thumbnailUrl: 'image.png',
+    mimeType: 'mimeType'
+};
+
+const file = new File([''], 'filename');
+const validity = {
+    valid: true,
+    fileTypeMismatch: false,
+    maxFileSizeExceeded: false,
+    multipleFilesDropped: false
+};
+
+const DROP_ZONE_FILE_EVENT: DropZoneFileEvent = {
+    file,
+    validity
+};
+
+describe('DotBinaryFieldComponent', () => {
+    let spectator: Spectator<DotBinaryFieldComponent>;
+    let store: DotBinaryFieldStore;
 
     const createComponent = createComponentFactory({
-        component: BinaryFieldComponent,
+        component: DotBinaryFieldComponent,
         imports: [
             NoopAnimationsModule,
             ButtonModule,
             DialogModule,
             MonacoEditorModule,
+            DotUiMessageComponent,
             HttpClientTestingModule
         ],
         providers: [
+            DotBinaryFieldStore,
+            {
+                provide: DotUploadService,
+                useValue: {
+                    uploadFile: ({ file }) => {
+                        return new Promise((resolve) => {
+                            if (file) {
+                                resolve(TEMP_FILE_MOCK);
+                            }
+                        });
+                    }
+                }
+            },
             {
                 provide: DotMessageService,
                 useValue: CONTENTTYPE_FIELDS_MESSAGE_MOCK
             }
         ]
     });
 
-    beforeEach(() => (spectator = createComponent()));
+    beforeEach(() => {
+        spectator = createComponent({
+            props: {
```

Comment:
if you use ` spectator.setInput({
className: 'danger'
});` that trigger automatic ngOnInit and OnChanges

Reply:
I set `detectChanges: false` and called `spectator.detectChanges();` where needed.